miasm2/arch/aarch64/arch.py

```diff
@@ -0,0 +1,1968 @@
+
+simdregs = (simd08_info.parser | simd16_info.parser |
+            simd32_info.parser | simd64_info.parser)
+simdregs_h = (simd32_info.parser | simd64_info.parser | simd128_info.parser)
+
+simdregs_h_zero = (simd32_info.parser |
+                   simd64_info.parser | simd128_info.parser | int_or_expr)
+
+
+def ast_id2expr(t):
+    if not t in mn_aarch64.regs.all_regs_ids_byname:
+        r = m2_expr.ExprId(asm_label(t))
+    else:
+        r = mn_aarch64.regs.all_regs_ids_byname[t]
+    return r
+
+
+def ast_int2expr(a):
+    return m2_expr.ExprInt64(a)
+
+gpregs_info = {32: gpregs32_info,
+               64: gpregs64_info}
+gpregsz_info = {32: gpregsz32_info,
+                64: gpregsz64_info}
+
+
+simds_info = {8: simd08_info,
+              16: simd16_info,
+              32: simd32_info,
+              64: simd64_info,
+              128: simd128_info}
+
+
+my_var_parser = parse_ast(ast_id2expr, ast_int2expr)
+base_expr.setParseAction(my_var_parser)
+
+
+def deref2expr_nooff(t):
+    t = t[0]
+    # XXX default
+    return m2_expr.ExprOp("preinc", t[0], m2_expr.ExprInt64(0))
+
+
+def deref2expr_post(t):
+    t = t[0]
+    if t[1] in regs_module.all_regs_ids:
+        raise StopIteration
+    return m2_expr.ExprOp("postinc", t[0], t[1])
+
+
+def deref2expr_pre(t):
+    t = t[0]
+    if t[1] in regs_module.all_regs_ids:
+        raise StopIteration
+    return m2_expr.ExprOp("preinc", t[0], t[1])
+
+
+def deref2expr_pre_wb(t):
+    t = t[0]
+    if t[1] in regs_module.all_regs_ids:
+        raise StopIteration
+    return m2_expr.ExprOp("preinc_wb", t[0], t[1])
+
+LBRACK = Suppress("[")
+RBRACK = Suppress("]")
+COMMA = Suppress(",")
+POSTINC = Suppress("!")
+
+deref_nooff = Group(
+    LBRACK + gpregs64_info.parser + RBRACK).setParseAction(deref2expr_nooff)
+deref_off_post = Group(LBRACK + gpregs64_info.parser +
+                       RBRACK + COMMA + int_or_expr64).setParseAction(deref2expr_post)
+deref_off_pre = Group(LBRACK + gpregs64_info.parser +
+                      COMMA + int_or_expr64 + RBRACK).setParseAction(deref2expr_pre)
+deref_off_pre_wb = Group(LBRACK + gpregs64_info.parser + COMMA +
+                         int_or_expr64 + RBRACK + POSTINC).setParseAction(deref2expr_pre_wb)
+
+deref = (deref_off_post | deref_off_pre_wb | deref_off_pre | deref_nooff)
+
+
+def deref_ext2op(t):
+    t = t[0]
+    if len(t) == 4:
+        expr = set_imm_to_size(t[1].size, t[3])
+        if expr is None:
+            raise StopIteration
+        return m2_expr.ExprOp('segm', t[0], m2_expr.ExprOp(t[2], t[1], expr))
+    elif len(t) == 2:
+        return m2_expr.ExprOp('segm', t[0], t[1])
+
+    fdsfds
+
+deref_ext2 = Group(LBRACK + gpregs_32_64 + COMMA + gpregs_32_64 +
+                   Optional(all_extend2_t + int_or_expr) + RBRACK).setParseAction(deref_ext2op)
+
+
+class additional_info:
+
+    def __init__(self):
+        self.except_on_instr = False
+        self.lnk = None
+        self.cond = None
+
+conds = [
```

Comment:
`conds`, `conds_inv`, `brcond`, `uints`, ... are constants. You may rewrite them in upper case, to clarify the code and distinguish them with vars.